src/secontext.c

```diff
@@ -9,6 +9,7 @@
 
 #include <stdlib.h>
 #include <fcntl.h>
+#include <libgen.h>
```

Comment:
I don't think using weird routines declared in this header is justified.

Reply:
That's for `dirname()` and `basename()`. I can reimplement these if needed.

Reply:
basename() from <string.h> is better.  I suppose this works:
```
#include <libgen.h>
#undef basename
```

Reply:
@ldv-alt What do you think of searching for the last slash from the end instead of using `dirname()` and `basename()`.
I was initially fearing that this would require handling escaping, but it seems on Linux at least that it's not possible to have a slash in the file name at all.
It may be on Windows (+ NFS or Samba), but in such case there won't be SELinux labels for sure, so the code will not execute at all.

Reply:
I think in this case we can assume that slash is a directory separator.